src/toGamut.js

```diff
@@ -163,40 +165,36 @@ export function toGamutCSS (origin, { space = origin.space }) {
 		const destColor = to(_color, space);
 		const spaceCoords = Object.values(space.coords);
 		destColor.coords = destColor.coords.map((coord, index) => {
-			const spaceCoord = spaceCoords[index];
-			if (("range" in spaceCoord)) {
-				if (coord < spaceCoord.range[0]) {
-					return spaceCoord.range[0];
-				}
-				if (coord > spaceCoord.range[1]) {
-					return spaceCoord.range[1];
-				}
+			if ("range" in spaceCoords[index]) {
+				const [min, max] =  spaceCoords[index].range;
+				return util.clamp(min, coord, max);
 			}
 			return coord;
 		});
 		return destColor;
 	}
 	let min = 0;
 	let max = origin_OKLCH.coords[1];
-
 	let min_inGamut = true;
-	let current;
+	let current = clone(origin_OKLCH);
+    let clipped = clip(current);
+
+    let E = deltaEOK(clipped, current);
+    if (E < JND) {
+		return clipped;
+	}
 
 	while ((max - min) > ε) {
 		const chroma = (min + max) / 2;
-		current = clone(origin_OKLCH);
 		current.coords[1] = chroma;
 		if (min_inGamut && inGamut(current, space)) {
```

Comment:
One thing that I missed, this should really be using an `epsilon` of zero. The earlier shortcut out should be based on an `epsilon` of zero as well (line 160).

Reply:
I updated this in https://github.com/color-js/color.js/pull/352/commits/907c8f3544cd6436287530fd8ce515b2fc2216b2

The [Gamut tests](https://deploy-preview-352--colorjs.netlify.app/test/?test=gamut) are passing on this branch- are there additional cases I should add for checking what you were looking at in #378 ?